oonidata/dataformat.py

```diff
@@ -30,6 +32,19 @@ class BinaryData:
 Failure = Optional[str]
 
 
+def guess_decode(s: bytes) -> str:
+    """
+    best effort decoding of a string of bytes
+    """
+    for encoding in ("ascii", "utf-8", "latin1"):
+        try:
+            return s.decode(encoding)
+        except UnicodeDecodeError:
+            pass
+    log.warning(f"unable to decode '{s}'")
+    return s.decode("ascii", "ignore")
```

Comment:
I am confused: this function used to be part of #2 and now I see it again?!

I merged the main branch into this branch but this addition is still here and I don't think it should be

Reply:
Ah, now it's more clear to me what happened: it has been moved 😬

Reply:
Yeah I moved it into here because I needed to use it as part of dataformat and wanted to avoid circular dependencies. I think overall the layout can be improved a bit with some better restructuring.